src/common/pa_memorybarrier.h

```diff
@@ -61,13 +61,24 @@
  ****************/
 
 #if defined(__APPLE__)
-#   include <libkern/OSAtomic.h>
-    /* Here are the memory barrier functions. Mac OS X only provides
-       full memory barriers, so the three types of barriers are the same,
-       however, these barriers are superior to compiler-based ones. */
-#   define PaUtil_FullMemoryBarrier()  OSMemoryBarrier()
-#   define PaUtil_ReadMemoryBarrier()  OSMemoryBarrier()
-#   define PaUtil_WriteMemoryBarrier() OSMemoryBarrier()
+/* Support for stdatomic was added in XCode 7.
+ * If you need to build on an older version, define PA_MAC_USE_OS_MEMORY_BARRIER.
+ * TODO Find an automated way to check for stdatomic support. */
```

Comment:
This can be handled automatically by the build system by checking if `stdatomic.h` exists: https://cmake.org/cmake/help/latest/module/CheckIncludeFile.html